### PR TITLE
Fix `terraform import` command. Add `terraform clean` command. Update spacelift processor

### LINKED ISSUE #70: Improve `terraform import` functionality
Have a question? Please checkout our [Slack Community](https://slack.cloudposse.com) or visit our [Slack Archive](https://archive.sweetops.com/). 

[Slack Community](https://slack.cloudposse.com)

## Describe the Feature

In order to import resources, the var file needs to be supplied.

```
atmos terraform import eks -s use2-corp \
  -var-file /components/terraform/eks/use2-corp-eks.terraform.tfvars.json \
  "module.eks_cluster.aws_eks_cluster.default[0]" \
  acme-use2-corp-eks-cluster
```

It would be nice if the var file can be automatically generated and supplied by atmos like this

```
atmos terraform import eks -s use2-corp \
  "module.eks_cluster.aws_eks_cluster.default[0]" \
  acme-use2-corp-eks-cluster
```

otherwise atmos isn't adding anything since I could do the above import by cd'ing directly into the component directory and run the terraform import manually with the supplied import var.

```
cd components/terraform/eks
terraform import \
  -var-file /components/terraform/eks/use2-corp-eks.terraform.tfvars.json \
  "module.eks_cluster.aws_eks_cluster.default[0]" \
  acme-use2-corp-eks-cluster
```

### PR DESCRIPTION
**Pull request by aknysh**
## what
* Fix `terraform import` command
* Add `terraform clean` command
* Update spacelift processor to use logical stack names and find all top-level stacks automatically by using `atmos.yaml` CLI config

## why
* `varfile` was generated, but `terraform import` did not use the generated `varfile` in `--var-file`
* `terraform clean` command deletes `varfile`, `planfile`, `.terraform` folder, '.terraform.lock.hcl' file, and the folder that `TF_DATA_DIR` ENV variable points to (usage: `atmos terraform clean <component> -s=<stack>`)
* Make the Spacelift processor use the logical stack names (derived from the context) to generate Spacelift stack names and Spacelift folders. Support stack and component folders, tenant (and all attributes from the context), and complex imports (Globs, folders in imports). Make the Spacelift processor use `atmos.yaml` CLI config to be able to find all top-level stacks automatically. Maintain backwards compatibility with the old Spacelift stack processor logic (where a list of stack config files was provided).

## references
* https://www.terraform.io/docs/cli/config/environment-variables.html#tf_data_dir

## related
* Closes #70  

## `terraform clean` test

```
Deleting '.terraform' folder
Deleting '.terraform.lock.hcl' file
Deleting terraform varfile: tenant1-ue2-dev-test-component-override.terraform.tfvars.json
Deleting terraform planfile: tenant1-ue2-dev-test-component-override.planfile
Found ENV var TF_DATA_DIR=./test
Do you want to delete the folder './test'? (only 'yes' will be accepted to approve)
Enter a value: yes
Deleting folder './test'
```

### CODE REVIEW

File: internal/exec/terraform.go [hunk 1]
```diff
@@ -49,6 +49,26 @@ func ExecuteTerraform(cmd *cobra.Command, args []string) error {
...
+
+	if info.SubCommand == "clean" {
+		fmt.Println(fmt.Sprintf("Deleting terraform varfile: %s", path.Join(componentPath, varFile)))
+		_ = os.Remove(path.Join(componentPath, varFile))
+
+		fmt.Println(fmt.Sprintf("Deleting terraform planfile: %s", path.Join(componentPath, planFile)))
+		_ = os.Remove(path.Join(componentPath, planFile))
+
+		fmt.Println("Deleting '.terraform' folder")
+		_ = os.RemoveAll(path.Join(componentPath, ".terraform"))
```
Comment (osterman) [inline internal/exec/terraform.go:63]: Or should it also check TF_WORK_DIR?

Comment (aknysh) [reply internal/exec/terraform.go:64]: added this check as well

File: internal/exec/terraform.go [hunk 2]
```diff
@@ -49,6 +49,33 @@ func ExecuteTerraform(cmd *cobra.Command, args []string) error {
 		))
 	}
 
+	varFile := fmt.Sprintf("%s-%s.terraform.tfvars.json", info.ContextPrefix, info.Component)
+	planFile := fmt.Sprintf("%s-%s.planfile", info.ContextPrefix, info.Component)
+
+	if info.SubCommand == "clean" {
+		tfDataDir := os.Getenv("TF_DATA_DIR")
+		if len(tfDataDir) > 0 {
+			color.Cyan("Found ENV var TF_DATA_DIR=%s", tfDataDir)
```
Comment (osterman) [inline internal/exec/terraform.go:58]: Let's make sure it's not `/` =)

Comment (aknysh) [reply internal/exec/terraform.go]: updated


```
Deleting '.terraform' folder
Deleting '.terraform.lock.hcl' file
Deleting terraform varfile: tenant1-ue2-dev-test-component-override.terraform.tfvars.json
Deleting terraform planfile: tenant1-ue2-dev-test-component-override.planfile
Found ENV var TF_DATA_DIR=./test
Do you want to delete the folder './test'? (only 'yes' will be accepted to approve)
Enter a value: yes
Deleting folder './test'
```

**Pull request merged**